test/recipes/95-test_external_pyca_data/cryptography.sh

```diff
@@ -44,10 +44,39 @@ pip install -U pip
 
 cd pyca-cryptography
 
+# Handling Rust
+export RUSTUP_HOME="$PWD/.cache/.rustup"
+export CARGO_HOME="$PWD/.cache/.cargo"
+export PATH="$CARGO_HOME/bin:$PATH"
+RUST_VERSION="1.85.1"
+
+# Install Rust
+if [ ! -f $CARGO_HOME/bin/cargo ]; then
+     echo "Installing Rust..."
+     curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y \
+         --no-modify-path \
+         --default-toolchain "$RUST_VERSION" \
+         --profile minimal
+fi
+
+# Load cargo/env
+. "$CARGO_HOME/env"
+
+# Explicitly set the default toolchain (in case installation didn't)
+rustup default "$RUST_VERSION" >/dev/null 2>&1
+
+# Verify installation
+if ! command -v cargo >/dev/null 2>&1; then
+    echo "Error: Rust installation failed!" >&2
+    exit 1
+fi
+
+echo "Rust is installed: $(rustc --version)"
```

Comment:
Should Rust be uninstalled after the tests like it's with the python virtual environment below?